wgpu-core/src/device/mod.rs

```diff
@@ -2116,6 +2121,20 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         let (mut buffer_guard, _) = hub.buffers.write(&mut token);
         let buffer = &mut buffer_guard[buffer_id];
 
+        match &buffer.map_state {
+            resource::BufferMapState::Unmapped => {
+                log::error!("Buffer already unmapped");
+                return;
+            },
+            resource::BufferMapState::Mapped => buffer.map_state = resource::BufferMapState::Unmapped,
+            resource::BufferMapState::Referenced {
+                map_pending: pending_op,
+            } => if pending_op.is_some() {
+                log::error!("Buffer mapping requested");
```

Comment:
https://gpuweb.github.io/gpuweb/#GPUBuffer-mapWriteAsync

> 5. Enqueue an operation on the Queue timeline that will execute the following:

Hmm, this sounds like it should allow a queue of mappings rather than canceling the previous one.

Reply:
spec is a WIP :)